### ISSUE TITLE
How to build using MSbuild(Visual Studio)?

### ISSUE THREAD

**Issue by Escapingbug** [neovim/neovim#10813]
I'm trying to build following [this instruction with msbuild](https://gist.github.com/equalsraf/3bd1da169de03c50ff0fe296191572af)

But I get error messages like this:

```
D:\reinstallable\sources\neovim-official\src\nvim\viml\parser\expressions.c(124,14): error C2059:
语法错误:“enum [tag]” [D:\reinstallable\sources\neovim-official\build\src\nvim\nvim.vcxproj]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
9,27): error C2061:  语法错误: 标识符“node_lvl” [D:\reinstallable\sources\neovim-official\build\src\
nvim\n
vim.vcxproj]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
9,27): error C2059:  语法错误:“;” [D:\reinstallable\sources\neovim-official\build\src\nvim\nvim.vcxp
roj
]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
9,54): error C2059:  语法错误:“<parameter-list>” [D:\reinstallable\sources\neovim-official\build\src
\nv
im\nvim.vcxproj]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
10,37): error C2061:  语法错误: 标识符“node_ass” [D:\reinstallable\sources\neovim-official\build\src
\nvim\
nvim.vcxproj]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
10,37): error C2059:  语法错误:“;” [D:\reinstallable\sources\neovim-official\build\src\nvim\nvim.vcx
pro
j]
D:\reinstallable\sources\neovim-official\build\src\nvim\auto\viml/parser/expressions.c.generated.h(
10,64): error C2059:  语法错误:“<parameter-list>” [D:\reinstallable\sources\neovim-official\build\sr
c\n
vim\nvim.vcxproj]
```

It reports identifier missing and grammar mistakes. I actually could build this before, but don't know why it is not working now. Need help on this one.

**Comment by justinmk** [comment]
https://gist.github.com/equalsraf/3bd1da169de03c50ff0fe296191572af could be outdated. Did you try https://github.com/neovim/neovim/wiki/Building-Neovim#windows--msvc ?

**Comment by Escapingbug** [comment]
I have read that documentation, but only to find instructions on using Visual Studio. What if I'd like to use only commandline tools like `MSBuild` to build the project?

**Comment by justinmk** [comment]
> use only commandline tools like `MSBuild` to build the project?

Best reference for that is [build.ps1](https://github.com/neovim/neovim/blob/master/ci/build.ps1) which is used on AppVeyor CI. 

@janlazo made an effort to generalize it for desktop use: https://github.com/neovim/neovim/pull/8405 (but the PR was not finished--would be great if someone adopted it)

**Comment by Escapingbug** [comment]
Hmm.. I'm still confused why I got that error. Since my operations should work by reading through that shell script. The only difference I see is the version: I'm using "Visual Studio 16 2019" instead of "Visual Studio 15 2017".

What might possibly be the reason that I was having that error message? I'm not sure how I can make this right...

**Comment by blueyed** [comment]
> It reports identifier missing and grammar mistakes.
> …
> The only difference I see is the version: I'm using "Visual Studio 16 2019" instead of "Visual Studio 15 2017".

IIRC there are a lot of warnings with MSVC 2017 on AppVeyor already, and therefore I could imagine that there are errors then with a newer version.

**Comment by Escapingbug** [comment]
But the weird thing is that I could build it some time ago(same version), but I didn't remember I touched anything.

What might help to resolve this situation.. I'm kind of stuck here..

**Comment by justinmk** [comment]
Start with the first error, and see if you can work around it. The first error, C2059, mentions `enum [tag]` on line 124. But that line is blank in the Neovim source. Are you sure  `viml\parser\expressions.c`  was not modified accidentally?

**Comment by Escapingbug** [comment]
Yes, I'm sure. I have already cloned a whole new repo. And yes, there really isn't anything on that line. This is pretty strange.

**Comment by Escapingbug** [comment]
Now I'm deleting all the repos I have, and will clone a new one, and will build after reboot. This is truly strange..

**Comment by Escapingbug** [comment]
No luck, still getting this error. Can someone else reproduce this?

**Comment by blueyed** [comment]
No Windows here..
It might be due to new code in Neovim?  (i.e. the error only happens since then for you?)

**Comment by Escapingbug** [comment]
Ok, I've checked out previous versions, not working. So it might be environment problem? Anyone got an idea? Also, I'm not sure how it can be done officially. I've already read the documentation, it says open the project. But there is no project until I use cmake to generate one, so I'm not sure if there is anything missing. But if I generates using cmake, then I get the exact same error as I got before. Those errors even have no related source code line!

**Comment by erw7** [comment]
@Escapingbug Will building succeed if you apply https://github.com/erw7/neovim/commit/ddb6c2dd6c8141c817055ab567b3eedbd4616b90. What happens if you build with the Windows display language changed to English?

**Comment by Escapingbug** [comment]
> 
> 
> @Escapingbug Will building succeed if you apply [erw7@ddb6c2d](https://github.com/erw7/neovim/commit/ddb6c2dd6c8141c817055ab567b3eedbd4616b90). What happens if you build with the Windows display language changed to English?

It works! Thanks!

For future reference, what I have done (I'm not sure which one is the exact reason):

* settings -> time and language -> language -> Windows Display Language sets to "English (United States)". However, this may not be the reason, since when I switched back, it still works.
* control panel -> time and region -> region -> administrative tab -> change system region -> change to "English (United States)" (This again may not be the reason, I have changed back, and still works)
* at the "change system region" part, select "beta: use Unicode UTF-8 for worldwide language support"
